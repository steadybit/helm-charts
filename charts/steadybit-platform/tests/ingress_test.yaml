templates:
  - ingress.yaml
chart:
  appVersion: 0.0.0
  version: 0.0.0
tests:
  - it: should disable ingress
    set:
      ingress:
        enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should enable ingress
    set:
      ingress:
        enabled: true
        annotations:
          this: that
        hosts:
          - example.steadybit.com
    asserts:
      - matchSnapshot: { }
  - it: should enable ingress in plain kubernetes
    asserts:
      - matchSnapshot: { }
  - it: should disable ingress in openshift
    capabilities:
      apiVersions:
        - route.openshift.io/v1
    asserts:
      - hasDocuments:
          count: 0
