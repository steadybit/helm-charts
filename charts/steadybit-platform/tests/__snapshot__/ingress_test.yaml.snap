should enable ingress:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        this: that
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      rules:
        - host: example.steadybit.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-steadybit-platform
                    port:
                      number: 7878
                path: /ws
                pathType: Prefix
              - backend:
                  service:
                    name: RELEASE-NAME-steadybit-platform
                    port:
                      number: 80
                path: /
                pathType: Prefix
should enable ingress in plain kubernetes:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      rules:
        - http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-steadybit-platform
                    port:
                      number: 7878
                path: /ws
                pathType: Prefix
              - backend:
                  service:
                    name: RELEASE-NAME-steadybit-platform
                    port:
                      number: 80
                path: /
                pathType: Prefix
